src/LibraryManager/Utilities/ParallelUtility.cs

```diff
@@ -34,19 +34,21 @@ public static async Task ForEachAsync<T>(Func<T, Task> action, int degreeOfParal
             items = items ?? throw new ArgumentNullException(nameof(items));
             cancellationToken.ThrowIfCancellationRequested();
 
-            var semaphore = new SemaphoreSlim(degreeOfParallelism, degreeOfParallelism);
-            var allTasks = new List<Task>();
-
-            foreach (T item in items)
+            using (var semaphore = new SemaphoreSlim(degreeOfParallelism, degreeOfParallelism))
```

Comment:
Not that it matters, much, but to simplify the edit (which I'm assuming was in response to an analyzer warning) could this have been a using declaration?

Reply:
It would simplify it but then I'd have to bump the compiler version.  Will do that soon (now that I've noticed haha), but when I do I want to make all the appropriate style changes at once for consistency.